services/infra-api/serverless.yml

```diff
@@ -48,6 +76,140 @@ resources:
+            AccessKey: ${self:custom.nrLicenseKey}
+          BufferingHints:
+            IntervalInSeconds: 60
+            SizeInMBs: 1
+          RetryOptions:
+            DurationInSeconds: 60
+          S3Configuration:
+            CompressionFormat: GZIP
+            BucketARN: !GetAtt S3FirehoseEventsBucket.Arn
+            RoleARN: !GetAtt FirehoseRole.Arn
+          RoleARN: !GetAtt FirehoseRole.Arn
+
+    MetricStreamRole:
+      Type: AWS::IAM::Role
+      Condition: CreateNRInfraMonitoring
+      Properties:
+        RoleName: NewRelicInfraMetricStreamRole
+        AssumeRolePolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Effect: Allow
+              Principal:
+                Service:
+                  - streams.metrics.cloudwatch.amazonaws.com
+              Action:
+                - 'sts:AssumeRole'
+              Condition:
+                StringEquals:
+                  'sts:ExternalId': !Sub ${AWS::AccountId}
+        Policies:
+          - PolicyName: MetricStreams-FirehosePutRecords
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - 'firehose:PutRecord'
+                    - 'firehose:PutRecordBatch'
+                  Resource: !GetAtt FirehoseStreamToNewRelic.Arn
+        Description: Role to allow a metric stream put metrics into a firehose
+
+    FirehoseRole:
+      Type: AWS::IAM::Role
+      Condition: CreateNRInfraMonitoring
+      Properties:
+        RoleName: NewRelicInfraFirehoseRole
+        AssumeRolePolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Effect: Allow
+              Principal:
+                Service:
+                  - firehose.amazonaws.com
+              Action:
+                - 'sts:AssumeRole'
+        Policies:
+          - PolicyName: Firehose-S3Access
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - 's3:AbortMultipartUpload'
+                    - 's3:GetBucketLocation'
+                    - 's3:GetObject'
+                    - 's3:ListBucket'
+                    - 's3:ListBucketMultipartUploads'
+                    - 's3:PutObject'
+                  Resource:
+                    - !GetAtt S3FirehoseEventsBucket.Arn
+                    - !Join ['', [!GetAtt S3FirehoseEventsBucket.Arn, '/*']]
+        Description: Role to allow firehose stream put events into S3 backup bucket
+
+    CloudWatchMetricStream:
+      Type: AWS::CloudWatch::MetricStream
+      Condition: CreateNRInfraMonitoring
+      Properties:
+        Name: ${self:custom.nrMetricStreamName}
+        FirehoseArn: !GetAtt FirehoseStreamToNewRelic.Arn
+        RoleArn: !GetAtt MetricStreamRole.Arn
+        OutputFormat: 'opentelemetry0.7'
+
```

Comment:
CloudWatch lets us transform this data, but they have OpenTelemetry support built in which is what NewRelic will use.